docs/articles/documentation/test-api/user-roles.md

```diff
@@ -0,0 +1,111 @@
+---
+layout: docs
+title: User Roles
+permalink: /documentation/test-api/user-roles.html
+checked: true
+---
+# User Roles
+
+Many test scenarios involve the activity of more than one user. TestCafe addresses these scenarios by providing a convenient way
+to isolate authentication test actions and apply them easily whenever you need to switch the user account.
+
+A piece of logic that logs in a particular user is called a *role*. Define a role for each user account participating in your test.
+
+## Creating and Using Roles
+
+Use the `Role` constructor to create and initialize a role.
+
+```text
+Role( func( t ) )
+```
+
+Parameter | Type     | Description
+--------- | -------- | --------------------------------------------------------------------------------
+`func`    | Function | An asynchronous function that contains logic that authenticates the user.
+`t`       | Object   | The [test controller](test-code-structure.md#test-controller) used to access test run API.
+
+```js
+import { Role } from 'testcafe';
+
+const regularAccUser = Role(async t => {
+    await t
+        .typeText('#login', 'TestUser')
+        .typeText('#password', 'testpass')
+        .click('#sign-in');
+});
+
+const facebookAccUser = Role(async t => {
+    await t
+        .click('#sign-in-with-facebook')
+        .typeText('#email', 'testuser@mycompany.com')
+        .typeText('#pass', 'testpass')
+        .click('#submit');
+});
+
+const admin = Role(async t => {
+    await t
+        .typeText('#login', 'Admin')
+        .typeText('#password', 'adminpass')
+        .click('#sign-in');
+});
+```
+
+> Role initialization code is executed only once when the role is used for the first time.
+
+After you created roles, you can switch between users at any moment. Use the `t.useRole` function for this.
+
+```text
+t.useRole( role )
```

Comment:
It's worth mention that Role are like switching between real users. Thus, if you already used role in the particular test after switching from it and back to it all cookies that were set for this role in test previously will persist. E.g. you have role of user authenticated on Facebook. You switch to this role and perform authentication for Gmail. Then you switch to another role and back to facebook user role. Facebook user roll will be authenticated both on Facebook on Gmail. However, if other test will use this role it only will be authenticated on Faceebook. So, figuratively, branch of role state is created for each test run.